thirdeye/thirdeye-spi/src/main/java/org/apache/pinot/thirdeye/detection/performance/PerformanceMetrics.java

```diff
@@ -0,0 +1,155 @@
+
+  @JsonProperty
+  private PerformanceMetric responseRate;
+
+  @JsonProperty
+  private PerformanceMetric precision;
+
+  @JsonProperty
+  private PerformanceMetric recall;
+
+  public PerformanceMetric getTotalAnomalies() { return totalAnomalies; }
+
+  public PerformanceMetric getResponseRate() { return responseRate; }
+
+  public PerformanceMetric getPrecision() { return precision; }
+
+  public PerformanceMetric getRecall() { return recall; }
+
+  /**
+   * Builder for performance metrics
+   */
+  public static class Builder {
+    private long respondedAnomalies, truePos, falsePos, falseNeg, numAnomalies = 0L;
+    private boolean includeTotalAnomalies, includeResponseRate, includePrecision, includeRecall = false;
+
+    private PerformanceMetric buildTotalAnomalies() {
+      PerformanceMetric totalAnomalies = new PerformanceMetric();
+      totalAnomalies.setValue((double) this.numAnomalies);
+      totalAnomalies.setType(PerformanceMetricType.COUNT);
+      return totalAnomalies;
+    }
+
+    private PerformanceMetric buildResponseRate() {
+      PerformanceMetric responseRate = new PerformanceMetric();
+      double rate = (double)this.respondedAnomalies / this.numAnomalies * 100;
+      responseRate.setValue(rate);
+      responseRate.setType(PerformanceMetricType.PERCENT);
+      return responseRate;
+    }
+
+    private PerformanceMetric buildPrecision() {
+      PerformanceMetric precision = new PerformanceMetric();
+      double prec = (double)this.truePos / (this.truePos + this.falsePos) * 100;
+      precision.setValue(prec);
+      precision.setType(PerformanceMetricType.PERCENT);
+      return precision;
+    }
+
+    private PerformanceMetric buildRecall() {
+      PerformanceMetric recall = new PerformanceMetric();
+      double rec = (double)this.truePos / (this.truePos + this.falseNeg) * 100;
+      recall.setValue(rec);
+      recall.setType(PerformanceMetricType.PERCENT);
+      return recall;
+    }
+
+    /***
+     * Builds the performance given a list of anomalies. When calculating entity anomalies, it counts the total number of anomalies
+     * only on the parent level but includes children anomalies when calculating the performance
+     * @param anomalies A list of anomalies
+     */
+    public Builder (List<MergedAnomalyResultDTO> anomalies) {
+      anomalies.stream()
+          .forEach(anomaly -> {
+            if (!anomaly.isChild()) {
+              this.numAnomalies++;
+            }
+
+            if (anomaly.getAnomalyResultSource() != null) {
+              if (AnomalyResultSource.USER_LABELED_ANOMALY.equals(anomaly.getAnomalyResultSource())) {
+                if (anomaly.getFeedback() == null || anomaly.getFeedback().getFeedbackType().isAnomaly()) {
+                  // NOTE: includes user-created anomaly without feedback as false negative by default
+                  this.falseNeg++;
+                  this.respondedAnomalies++;
+                  return;
+                }
```

Comment:
There can be an edge case that the user labeled a reported anomaly as "NOT_ANOMALY", in that case, we still want to make `respondedAnomalies++`.